### PR TITLE
Inject node environment when executing a docker command (Allows DOCKE…

### PR DESCRIPTION
**Pull request by Vlatombe**
…R_HOST to be defined in Node Properties)

@reviewbybees

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/Docker.java [hunk 1]
```diff
@@ -69,12 +71,16 @@ public boolean hasImage(String image) throws IOException, InterruptedException {
...
         int status = launcher.launch()
-                .envs(dockerEnv.env())
+                .envs(getEnvVars())
                 .cmds(args)
                 .stdout(out).stderr(err).quiet(!verbose).join();
         return status == 0;
     }
 
+    private EnvVars getEnvVars() throws IOException, InterruptedException {
+        return new EnvVars(build.getEnvironment(listener)).overrideAll(dockerEnv.env());
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/Docker.java]: Does it make sense to cache the result?

File: src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/DockerDecoratedLauncher.java [hunk 2]
```diff
@@ -37,7 +37,11 @@ public Proc launch(ProcStarter starter) throws IOException {
         // Do not decorate launcher until SCM checkout completed
         if (!runInContainer.isEnabled()) return super.launch(starter);
 
-        runInContainer.getDocker().executeIn(runInContainer.container, starter);
+        try {
+            runInContainer.getDocker().executeIn(runInContainer.container, starter);
+        } catch (InterruptedException e) {
+            throw new IOException("Caught InterruptedException", e);
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/DockerDecoratedLauncher.java]: InterruptedException is supposed to interrupt the build with a single click. BTW there's no chance to properly propagate the issue here due to the API.

Comment (Vlatombe) [inline src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/DockerDecoratedLauncher.java]: yes unfortunately :/

File: src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/Docker.java (hunk 1 above)
Comment (Vlatombe) [inline src/main/java/com/cloudbees/jenkins/plugins/docker_build_env/Docker.java]: It doesn't hurt to.

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by amuniz** [comment]
:bee:

**Pull request merged**